stackrox/stackrox
# fix(ui): Fix accessibility issues for Alert heading level plus lint

### Description

Apply 80/20 principle to fix generic issues so team members can divide and conquer pages that have specific issues.

More like 100/1 principle: 114 changed files!

### Problem

> Heading levels should only increase by one

```html
<h4 class="pf-v5-c-alert__title">
```

### Analysis

PatternFly `Alert` element renders `h4` element by default.
* The content seems separate from heading hierarchy.
* To add insult to injury, `title` prop is sometimes the body, not a heading.

### Specific solution

It is often not obvious from component that renders `Alert` element what is the heading hierarchy on the page.

Because it is unlikely that screen reader users will consider it intuitive for heading levels to vary from page to page, specify `component="p"` prop to remove alert from heading hierarchy. I started with `component="div"` recently, but decided after reflection and research that this is content.

PatternFly renders `title` content with bold font weight and text size that is intuitive for visual users.

Given so much inconsistency in order of props, append `component="p"` prop immediately following `title` prop:
* For reviewability now.
* For pattern later.

### Generic solution

Prevent future problems via project-specific lint rules.

Adapt excellent exploration by **David Vail** in #10839

I used astexplorer dot net with typescript-eslint parser to find relevant properties for the rule.

### User-facing documentation


### Testing


#### Automated testing

1. `yarn lint` in ui/apps/platform
2. `yarn build` in ui/apps/platform and then compute branch - master for the following
    * `wc build/static/js/*.js`
        main.js 98 = 4615582 - 4615484
        total 2035 = 11700801 - 11698766
    * `ls -al build/static/js/*.js | wc`
        files 0 = 173 - 173
3. `yarn start` in ui/apps/platform

#### Manual testing

1. Visit /main/access-control/auth-providers

    * Before changes, see presence of accessibility issues.
        ![access-control_auth-providers_with_issue](https://github.com/user-attachments/assets/06f37f99-c3e2-4653-adb9-0bc6c05e13c6)

    * After changes, see absence of accessibility issues.
        ![access-control_auth-providers_without_issue](https://github.com/user-attachments/assets/a14f5cb1-bdd1-4941-bc17-a661e30cf69f)
